src/devices/vdml_rotation.c

```diff
@@ -73,6 +75,26 @@ int32_t rotation_set_reversed(uint8_t port, bool value) {
 	return_port(port - 1, 1);
 }
 
+int32_t rotation_init_with_reverse_flag(uint8_t port, bool reverse_flag) {
+	claim_port_i(port - 1, E_DEVICE_ROTATION);
+    uint16_t timeoutCount = 0;
+    // releasing mutex so vexBackgrounProcessing can run without being blocked.
+    do {
+        port_mutex_give(port - 1);
+        task_delay(5);
+        timeoutCount += 5;
+        claim_port_i(port - 1, E_DEVICE_ROTATION);
+        if (timeoutCount >= ROTATION_RESET_TIMEOUT) {
+            port_mutex_give(port - 1);
+            errno = EAGAIN;
+            return PROS_ERR;
+        }
+        device = device; // suppressing compiler warning
+    } while(vexDeviceAbsEncStatusGet(device->device_info) == 0);
+    vexAbsEncReverseFlagSet(port - 1, reverse_flag);
```

Comment:
We should be setting the flag before waiting for the flag status to update properly.